packages/contracts-bedrock/src/dispute/AnchorStateRegistry.sol

```diff
@@ -23,105 +22,222 @@ import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
-    struct StartingAnchorRoot {
-        GameType gameType;
-        OutputRoot outputRoot;
-    }
+    error InvalidGame();
+    error AnchorGameIsNewer(uint256 anchorGameL2BlockNumber, uint256 candidateGameL2BlockNumber);
+
+    event AnchorGameSet(IDisputeGame newAnchorGame);
+    event RespectedGameTypeSet(GameType gameType);
+    event DisputeGameBlacklisted(IDisputeGame game);
+    event GameRetirementTimestampSet(uint64 timestamp);
 
     /// @notice Semantic version.
+    // TODO: update?
     /// @custom:semver 2.0.1-beta.6
     string public constant version = "2.0.1-beta.6";
 
     /// @notice DisputeGameFactory address.
-    IDisputeGameFactory internal immutable DISPUTE_GAME_FACTORY;
+    IDisputeGameFactory public disputeGameFactory;
+
+    /// @notice The delay between when a dispute game is resolved and when it can be considered finalized.
+    uint256 public disputeGameFinalityDelaySeconds;
 
-    /// @notice Returns the anchor state for the given game type.
-    mapping(GameType => OutputRoot) public anchors;
+    // TODO: storage slot issue
+    // / @notice Returns the anchor state for the given game type.
+    // mapping(GameType => OutputRoot) public anchors;
+
+    // TODO: determine if this belongs
+    // uint256 __gap0;
 
     /// @notice Address of the SuperchainConfig contract.
     ISuperchainConfig public superchainConfig;
 
-    /// @param _disputeGameFactory DisputeGameFactory address.
-    constructor(IDisputeGameFactory _disputeGameFactory) {
-        DISPUTE_GAME_FACTORY = _disputeGameFactory;
+    /// @notice Returns whether a game is blacklisted.
+    mapping(IDisputeGame => bool) public isGameBlacklisted;
+
+    uint64 public gameRetirementTimestamp;
+
+    GameType public respectedGameType;
+
+    IFaultDisputeGame[] public anchorGameCandidates;
+
+    uint256 internal _anchorGameCandidateIndex;
+
+    IFaultDisputeGame internal _anchorGame;
+
+    uint256 public tryUpdateAnchorStateGas;
+
+    constructor() {
         _disableInitializers();
     }
 
     /// @notice Initializes the contract.
-    /// @param _startingAnchorRoots An array of starting anchor roots.
+    /// @param _disputeGameFactory DisputeGameFactory address.
+    /// @param _disputeGameFinalityDelaySeconds The delay between when a dispute game is resolved and when it can be
+    /// considered finalized.
+    /// @param _authorizedGame An authorized dispute game.
     /// @param _superchainConfig The address of the SuperchainConfig contract.
+    /// @param _tryUpdateAnchorStateGas The approximate gas limit for the tryUpdateAnchorState loop.
     function initialize(
-        StartingAnchorRoot[] memory _startingAnchorRoots,
-        ISuperchainConfig _superchainConfig
+        IDisputeGameFactory _disputeGameFactory,
+        uint256 _disputeGameFinalityDelaySeconds,
+        IFaultDisputeGame _authorizedGame,
+        ISuperchainConfig _superchainConfig,
+        uint256 _tryUpdateAnchorStateGas
```

Comment:
perhaps we could make this a constant; on the other hand, perhaps we need a setter